src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/InterceptedProjectProperties.cs

```diff
@@ -98,5 +110,34 @@ public void SetRuleContext(Rule rule)
             var ruleAwareProperties = DelegatedProperties as IRuleAwareProjectProperties;
             ruleAwareProperties?.SetRuleContext(rule);
         }
+
+        private IInterceptingPropertyValueProvider? GetFilteredProvider(string propertyName, (List<Lazy<IInterceptingPropertyValueProvider, IInterceptingPropertyValueProviderMetadata>> Providers, bool Filtered) valueTuple)
+        {
+            if (valueTuple.Filtered)
+            {
+                return ReturnProviderAfterFiltering();
+            }
+
+            valueTuple.Providers.RemoveAll(lazyProvider =>
+            {
+                string? appliesToExpression = lazyProvider.Value.GetType().GetCustomAttributes(true).OfType<AppliesToAttribute>().FirstOrDefault()?.AppliesTo;
+
+                return appliesToExpression is not null && !_project.Capabilities.AppliesTo(appliesToExpression);
+            });
+
+            valueTuple.Filtered = true;
```

Comment:
I don't think this will work. Value tuples are passed by value, so modifying the value here will not stick. It's only modifying a copy, and the original source remains unchanged.

My earlier comment about pulling this out to its own type stands, only that type would become a `class` and the property made read/write.